mmengine/fileio/backends/oss_backend.py

```diff
@@ -0,0 +1,809 @@
+            bucket_name (str): bucket name of OSS
+
+        Returns:
+            Bucket: An instance of OSS.
+        """
+
+        bucket = self.oss2.Bucket(self._auth, endpoint, bucket_name)
+        return bucket
+
+    def get(self, filepath: Union[str, Path]) -> bytes:
+        """Read bytes from a given ``filepath`` with 'rb' mode.
+
+        Args:
+            filepath (str or Path): Path to read data.
+
+        Returns:
+            bytes: Return bytes read from filepath.
+
+        Examples:
+            >>> backend = OSSBackend()
+            >>> filepath = 'oss://endpoint/bucket/file'
+            >>> backend.get(filepath)
+            b'hello world'
+        """
+        endpoint, bucket_name, obj_name = self._parse_path(filepath)
+        bucket = self._bucket_instance(endpoint, bucket_name)
+        object_stream = bucket.get_object(obj_name)
+        return object_stream.read()
+
+    def get_text(
+        self,
+        filepath: Union[str, Path],
+        encoding: str = 'utf-8',
+    ) -> str:
+        """Read text from a given ``filepath`` with 'r' mode.
+
+        Args:
+            filepath (str or Path): Path to read data.
+            encoding (str): The encoding format used to open the ``filepath``.
+                Defaults to 'utf-8'.
+
+        Returns:
+            str: Expected text reading from ``filepath``.
+
+        Examples:
+            >>> backend = OSSBackend()
+            >>> filepath = 'oss://endpoint/bucket/file'
+            >>> backend.get_text(filepath)
+            'hello world'
+        """
+        return str(self.get(filepath), encoding=encoding)
+
+    def put(self, obj: bytes, filepath: Union[str, Path]) -> None:
+        """Write bytes to a given ``filepath``.
+
+        Args:
+            obj (bytes): Data to be saved.
+            filepath (str or Path): Path to write data.
+
+        Examples:
+            >>> backend = OSSBackend()
+            >>> filepath = 'oss://endpoint/bucket/file'
+            >>> backend.put(b'hello world', filepath)
+        """
+        endpoint, bucket_name, obj_name = self._parse_path(filepath)
+        bucket = self._bucket_instance(endpoint, bucket_name)
+        bucket.put_object(obj_name, obj)
+
+    def put_text(
+        self,
+        obj: str,
+        filepath: Union[str, Path],
+        encoding: str = 'utf-8',
+    ) -> None:
+        """Write text to a given ``filepath``.
+
+        Args:
+            obj (str): Data to be written.
+            filepath (str or Path): Path to write data.
+            encoding (str): The encoding format used to encode the ``obj``.
+                Defaults to 'utf-8'.
+
+        Examples:
+            >>> backend = OSSBackend()
+            >>> filepath = 'oss://endpoint/bucket/file'
+            >>> backend.put_text('hello world', filepath)
+        """
+        self.put(bytes(obj, encoding=encoding), filepath)
+
+    def exists(self, filepath: Union[str, Path]) -> bool:
```

Comment:
This method will return True if filepath exists which can be a directory or file path.